scripts/serve/serve-static.js

```diff
@@ -1,177 +1,57 @@
-
-      if (fileExt === '.html') {
-        let fileContent = fs.readFileSync(filepath, 'utf8');
-        fileContent = setSiteLocale(fileContent, locale);
-        res.send(fileContent);
-      } else {
-        res.send(fs.readFileSync(filepath));
-      }
-    } else {
+  const options = {
+    ...index && { index },
+    // If an extensionless file can't be found search for it with .html and .htm
+    extensions: ['html', 'htm'],
+  };
+  // Return any files in the site. If no extension is provided check the file with the htm or html extension.
+  app.use(express.static(site, options));
+  // Match on *.htm, *.html or routes without extensions.
+  app.use([/\/[^.]*$/, '/*.html?'], (req, res, next) => {
+    // Return 404.html if provided.
+    res.status(404).sendFile('/404.html', { root: site }, () => {
+      // If there is an error, bail.
       next();
-    }
-  });
-
-  if (verbose) {
-    app.use((req, res, next) => {
-      const err = Logger.error(`Not Found: ${req.originalUrl}`, { context });
-      err.status = 404;
-      next(err);
     });
+  });
 
-    // error handler
-    app.use((err, req, res) => {
-      // set locals, only providing error in development
-      res.locals.message = err.message;
-      res.locals.error = err;
-
-      // render the error page
-      res.status(err.status || 500);
-      res.render('error');
-    });
-  }
-
-  return Promise.resolve(app);
-};
-
-// Setup a static express server
-const staticApp = (site) => {
-  const app = express();
-  app.use(express.static(site));
-  return Promise.resolve(app);
-};
-
-// Setup an app for either a virtual site or a static site. If a file system is not provided, serve the static site.
-const serveSite = (site, fs, index, locale, verbose) => {
-  if (fs) {
-    return virtualApp(site, index, locale, fs, verbose);
-  }
-
-  return staticApp(site);
+  return app;
 };
 
 // Generate a site if not provided and spin up an express server to serve the site.
 const serve = (options) => {
   const {
-    site, config, port, disk, index, locale, production, host, verbose,
+    // Setting defaults here instead of in cli.
+    site, port = 8080, host = '0.0.0.0', index,
   } = options;
 
-  const appPort = port || 8080;
-  const appIndex = index || 'index.html';
-  const appLocale = locale || process.env.LOCALE || 'en';
+  const app = staticApp(site, index);
 
-  return generateSite(site, config, disk, production)
-    .then(([sitePath, fs]) => serveSite(sitePath, fs, appIndex, appLocale, verbose))
-    .then((app) => {
-      const server = app.listen(appPort, host);
-      Logger.log(`Server started listening at port:${appPort}`, { context });
-      return server;
+  return new Promise((resolve, reject) => {
+    const server = app.listen(port, host, (err) => {
+      if (err) {
+        reject(err);
+      }
+      const localAddress = `http://${host}:${port}/`;
+      const networkAddress = `http://${ip.address()}:${port}/`;
```

Comment:
This network address might need to be pass from serveStatic service to make sense in CI runs